syntax = "proto2";

enum bleCommands {
  ACK = 0;
  NACK = 1;
  RESTART = 2;
  EREASE_WIFI_LIST = 3;
  ADD_WIFI = 4;
  REMOVE_WIFI = 5;
  SET_NICK = 6;
}

message wifiNetwork {
  required string ssid = 1;
  optional string password = 2;
}

message configuration {
  optional string nickName = 1;
  repeated wifiNetwork networks = 2;
}

message bleCommand  {
  required bleCommands command = 1;
  optional wifiNetwork nwt = 2;
  optional string nickName = 3;
}
